arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json

```diff
@@ -0,0 +1,525 @@
+                }
+            },
+            "description": "An Activity Log alert condition that is met when all its member conditions are met.",
+            "required": [
+                "allOf"
+            ]
+        },
+        "ActivityLogAlertLeafCondition": {
+            "properties": {
+                "field": {
+                    "type": "string",
+                    "description": "The name of the field that this condition will examine. The possible values for this field are (case-insensitive): 'resourceId', 'category', 'caller', 'level', 'operationName', 'resourceGroup', 'resourceProvider', 'status', 'subStatus', 'resourceType', or anything beginning with 'properties.'."
+                },
+                "equals": {
+                    "type": "string",
+                    "description": "The field value will be compared to this value (case-insensitive) to determine if the condition is met."
+                }
+            },
+            "description": "An Activity Log alert condition that is met by comparing an activity log field and value.",
+            "required": [
+                "field",
+                "equals"
+            ]
+        },
+        "ActivityLogAlertActionList": {
+            "description": "A list of activity log alert actions.",
+            "properties": {
+                "actionGroups": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/ActivityLogAlertActionGroup"
+                    },
+                    "description": "The list of activity log alerts."
+                }
+            }
+        },
+        "ActivityLogAlertActionGroup": {
+            "description": "A pointer to an Azure Action Group.",
+            "properties": {
+                "actionGroupId": {
+                    "type": "string",
+                    "description": "The resourceId of the action group. This cannot be null or empty."
+                },
+                "webhookProperties": {
+                    "additionalProperties": {
+                        "type": "string"
+                    },
+                    "description": "the dictionary of custom properties to include with the post operation. These data are appended to the webhook payload."
+                }
+            },
+            "required": [
+                "actionGroupId"
+            ]
+        },
+        "ErrorResponse": {
+            "description": "Describes the format of Error response.",
+            "type": "object",
+            "properties": {
+                "code": {
+                    "description": "Error code",
+                    "type": "string"
+                },
+                "message": {
+                    "description": "Error message indicating why the operation failed.",
+                    "type": "string"
+                }
+            }
+        },
+        "ActivityLogAlertResourcePatch": {
```

Comment:
This really isn't a resource object, looks like a simple request parameter. Instead of an allOf on Resource, consider setting location as a required property (if needed)
The reason I suggest is because ```ActivityLogAlertResourcePatch``` will be otherwise considered as a resource itself by our validation rules.